### PR TITLE
Support for multi-tenancy in a single namespace

### PR DESCRIPTION
**Pull request by sivanov-nuodb**
**Changes**

This change enables the installation of multiple domains and databases in a single Kubernetes namespace. Previously it was not possible to have two `admin` releases or two `database` primary releases with the same database name collocated in a single Kubernetes namespace. To prevent conflicts regarding resource naming, the names of existing resources have been changed (see the table below for details). 
Added needed plumbing so that user-defined resource labels can be injected into all provisioned Kubernetes resources. This can be used for easy selection of resources that are part of a specific domain or database. 

Type | Name | Owner Chart | Used by | New Name | Notes
-- | -- | -- | -- | -- | --
ConfigMap | waitscript | admin | LB Job | {{ template "admin.fullname" . }}-waitscript |  
ConfigMap | {{ .Values.database.name }}-restore | database | nuosm / nuorestore | {{ .Values.admin.domain }}-{{ .Values.database.name }}-restore |  
CronJob | full-hotcopy-{{ .Values.database.name }}-{{ $backupGroup }} | database | N/A | full-hotcopy-{{ .Values.admin.domain }}-{{ .Values.database.name }}-{{ $backupGroup }} |  
CronJob | incremental-hotcopy-{{ .Values.database.name }}-{{ $backupGroup }} | database | N/A | incremental-hotcopy-{{ .Values.admin.domain }}-{{ .Values.database.name }}-{{ $backupGroup }} |  
CronJob | journal-hotcopy-{{ .Values.database.name }}-{{ $backupGroup }} | database | N/A | journal-hotcopy-{{ .Values.admin.domain }}-{{ .Values.database.name }}-{{ $backupGroup }} |  
Secret | {{ .Values.database.name }}.nuodb.com | database | nuosm / nuorestore / ycsb | {{ .Values.admin.domain }}-{{ .Values.database.name }} | This secret is used internally only because it doesn't contain the database URL and it is not updated when the DBA password is changed.
Service | {{ .Values.database.name }}-{{ default .Values.admin.serviceSuffix.clusterip .Values.database.serviceSuffix.clusterip }} | database | direct SQL clients | This service is deprecated. | It can be enabled optionally by setting  `database.legacy.directService.enabled` value.
Headless Service | {{ .Values.database.name }} | database | none | The service is deprecated. | It can be enabled optionally by setting `database.legacy.headlessService.enabled` value.
Service | {{ .Values.admin.domain }}-{{ .Values.admin.serviceSuffix.balancer }} | database | N/A | {{ template "database.fullname" . }}-{{ default .Values.admin.serviceSuffix.balancer .Values.database.serviceSuffix.balancer }} | Not used for inter cluster connectivity.
Service | {{ .Values.admin.domain }}-{{ .Values.admin.serviceSuffix.nodeport }} | database | N/A | {{ template "database.fullname" . }}-{{ default .Values.admin.serviceSuffix.nodeport .Values.database.serviceSuffix.nodeport }} | Not used for inter cluster connectivity.
Job | restore-{{ $restoreTarget }} | restore | N/A | {{ template "restore.fullname" . }} |  

**Deprecated**

- The database headless service with name `{{ .Values.database.name }}`.
- The database direct `clusterIP` service with name `{{ .Values.database.name }}-clusterip`.

**Testing**

- regression testing
- new integration tests for resource labels
- moved `TestKubernetesMultiTenantDatabase` test in a separate file
- new E2E test

### CODE REVIEW

File: test/minikube/minikube_base_database_test.go [hunk 1]
```diff
@@ -163,71 +164,29 @@ func TestKubernetesBasicDatabase(t *testing.T) {
...
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.labels.cloud":              LABEL_CLOUD,
+			"database.te.labels.region":             LABEL_REGION,
+			"database.te.labels.zone":               LABEL_ZONE,
+			"database.sm.labels.cloud":              LABEL_CLOUD,
+			"database.sm.labels.region":             LABEL_REGION,
+			"database.sm.labels.zone":               LABEL_ZONE,
+		},
```
Comment (sivanov-nuodb) [inline test/minikube/minikube_base_database_test.go:182]: The `startDatabaseStatefulSet` subtest is unchanged but only made part of the main test. The `startDatabaseStatefulSetMultiTenant` has been extracted to separate test file.
Maybe it's easy to review this file side-by-side.

File: test/minikube/minikube_domain_resync_test.go [hunk 2]
```diff
@@ -70,55 +69,6 @@ func verifyProcessLabels(t *testing.T, namespaceName string, adminPod string) (a
 	return archiveVolumeClaims
 }
 
-func verifyLoadBalancer(t *testing.T, namespaceName string, adminPod string, deploymentOptions map[string]string) {
```
Comment (sivanov-nuodb) [inline test/minikube/minikube_domain_resync_test.go:73]: This function is moved to `test/minikube/verify_utility.go` file as it is now reused between files with different build tags.

File: test/minikube/minikube_base_database_test.go [hunk 3]
```diff
@@ -1,3 +1,4 @@
+//go:build short
```
[the remark this answers is not in the document]
Comment (sivanov-nuodb) [reply test/minikube/minikube_base_database_test.go:1]: This got automatically added by VSC editor.

**Review by adriansuarez — APPROVED** [review]
Looks good. Thanks for making this easy to review with the detailed summary.

**Review by bkelly-ndb — APPROVED** [review]

**Pull request merged**